api/src/main/java/com/javadiscord/jdi/internal/api/DiscordRequestBuilder.java

```diff
@@ -130,4 +131,41 @@ protected void setSuccessResult(DiscordResponse result) {
     protected void setFailureError(Throwable ex) {
         future.setException(ex);
     }
+
+    @Override
+    public String toString() {
+        String headersString =
+            headers.entrySet().stream()
+                .map(entry -> entry.getKey() + "=" + entry.getValue())
+                .collect(Collectors.joining(", "));
+
+        String queryParamsString =
+            queryParameters.entrySet().stream()
+                .map(entry -> entry.getKey() + "=" + entry.getValue())
+                .collect(Collectors.joining(", "));
+
+        String bodyString;
+        try {
+            bodyString = OBJECT_MAPPER.writeValueAsString(body);
+        } catch (Exception e) {
+            bodyString = "Error converting body to string: " + e.getMessage();
+        }
+
+        return this.getClass().getName()
+            + "{"
+            + "method="
+            + method
+            + ", path='"
+            + path
+            + '\''
+            + ", headers={"
+            + headersString
+            + '}'
+            + ", queryParameters={"
+            + queryParamsString
+            + '}'
+            + ", body="
+            + bodyString
+            + '}';
```

Comment:
Why not using proper string formatting?

Reply:
Example please? :)

Reply:
Is this not a thing for `toString` methods:
```java
"%s{method=%s, path='%s', headers={%s}, queryParameters={%s}, body=%s}".formatted(...)
```

Reply:
You can, the problem with doing it that way is when you have loads of fields like the one I've got, the initial string will be really long. 

I suppose the best way will be to use a StringBuilder - I can make this change.

Reply:
I tried StringBuilder and Intellij immediately said "hey use string" and reverted back to the String version that we currently have. Happy to leave it as it? @SquidXTV

Reply:
yeah im fine with that, though I dont really understand why string is better